tracim/views/contents_api/threads_controller.py

```diff
@@ -0,0 +1,197 @@
+    @hapic.handle_exception(WorkspaceNotFound, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(AuthenticationFailed, HTTPStatus.BAD_REQUEST)
+    @require_workspace_role(UserRoleInWorkspace.CONTRIBUTOR)
+    @require_content_types([thread_type])
+    @hapic.input_path(WorkspaceAndContentIdPathSchema())
+    @hapic.input_body(ThreadModifySchema())
+    @hapic.output_body(ThreadContentSchema())
+    def update_thread(self, context, request: TracimRequest, hapic_data=None):
+        """
+        update thread
+        """
+        app_config = request.registry.settings['CFG']
+        api = ContentApi(
+            current_user=request.current_user,
+            session=request.dbsession,
+            config=app_config,
+        )
+        content = api.get_one(
+            hapic_data.path.content_id,
+            content_type=ContentType.Any
+        )
+        with new_revision(
+                session=request.dbsession,
+                tm=transaction.manager,
+                content=content
+        ):
+            api.update_content(
+                item=content,
+                new_label=hapic_data.body.label,
+                new_content=hapic_data.body.raw_content,
+
+            )
+            api.save(content)
+        return api.get_content_in_context(content)
+
+    @hapic.with_api_doc(tags=[THREAD_ENDPOINTS_TAG])
+    @hapic.handle_exception(NotAuthenticated, HTTPStatus.UNAUTHORIZED)
+    @hapic.handle_exception(InsufficientUserWorkspaceRole, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(WorkspaceNotFound, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(AuthenticationFailed, HTTPStatus.BAD_REQUEST)
+    @require_workspace_role(UserRoleInWorkspace.READER)
+    @require_content_types([thread_type])
+    @hapic.input_path(WorkspaceAndContentIdPathSchema())
+    @hapic.output_body(ThreadRevisionSchema(many=True))
+    def get_thread_revisions(self, context, request: TracimRequest, hapic_data=None):  # nopep8
+        """
+        get thread revisions
+        """
+        app_config = request.registry.settings['CFG']
+        api = ContentApi(
+            current_user=request.current_user,
+            session=request.dbsession,
+            config=app_config,
+        )
+        content = api.get_one(
+            hapic_data.path.content_id,
+            content_type=ContentType.Any
+        )
+        revisions = content.revisions
+        return [
+            api.get_revision_in_context(revision)
+            for revision in revisions
+        ]
+
+    @hapic.with_api_doc(tags=[THREAD_ENDPOINTS_TAG])
+    @hapic.handle_exception(NotAuthenticated, HTTPStatus.UNAUTHORIZED)
+    @hapic.handle_exception(InsufficientUserWorkspaceRole, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(WorkspaceNotFound, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(AuthenticationFailed, HTTPStatus.BAD_REQUEST)
+    @require_workspace_role(UserRoleInWorkspace.CONTRIBUTOR)
+    @require_content_types([thread_type])
```

Comment:
This decorator may raise a `ContentTypeNotAllowed`. This exception looks like not being catched.
By the way, the test cases seem not to test this error case

Reply:
note : this remark is valid for every endpoint

Reply:
That's known: https://github.com/tracim/tracim/issues/605 . 
ContentTypeNotAllowed should raise which http code ?